src/plugins/usage_collection/server/usage_counters/rollups/rollups.ts

```diff
@@ -8,35 +8,39 @@
 
 import moment from 'moment';
 import type { ISavedObjectsRepository, Logger } from '@kbn/core/server';
-
-import {
-  type UsageCountersSavedObject,
-  USAGE_COUNTERS_SAVED_OBJECT_TYPE,
-} from '@kbn/usage-collection-plugin/server';
 import { USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS } from './constants';
-import { isSavedObjectOlderThan } from '../../common/saved_objects';
+import { type UsageCountersSavedObjectAttributes, USAGE_COUNTERS_SAVED_OBJECT_TYPE } from '..';
+import { isSavedObjectOlderThan } from '../saved_objects';
+import type { GetUsageCounter } from '../types';
 
-export async function rollUsageCountersIndices(
-  logger: Logger,
-  savedObjectsClient?: ISavedObjectsRepository
-) {
-  if (!savedObjectsClient) {
+export async function rollUsageCountersIndices({
+  logger,
+  usageCounters,
+  internalRepository,
+}: {
+  logger: Logger;
+  usageCounters: GetUsageCounter;
+  internalRepository?: ISavedObjectsRepository;
+}) {
+  if (!internalRepository) {
     return;
   }
 
   const now = moment();
 
   try {
     const { saved_objects: rawUiCounterDocs } =
-      await savedObjectsClient.find<UsageCountersSavedObject>({
+      await internalRepository.find<UsageCountersSavedObjectAttributes>({
         type: USAGE_COUNTERS_SAVED_OBJECT_TYPE,
         namespaces: ['*'],
         perPage: 1000, // Process 1000 at a time as a compromise of speed and overload
       });
```

Comment:
hmm, I don't think this approach is valid anymore...

If we have 12 dashboards viewed every day during the last 90 days, they'll show up in the first page and we won't remove others...

I wonder if we should change this to group per retention days (`domainId: 'dashboard' and updated_at < retentionPeriodDays` vs. `not domainId: 'dashboard' and updated_at < USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS`).

Reply:
Good catch! we'll have to think about a better strategy.

Reply:
After discussion, we can do a search by domainId, and filter by `updated_at < (now - retentionPeriodDays)`.
We can then loop through the different domain IDs.
Will assess `bulkDelete` following @TinaHeiligers's [latest comment](https://github.com/elastic/kibana/pull/187665#discussion_r1684950657).

Reply:
@afharo updated with https://github.com/elastic/kibana/pull/187665/commits/357b474909a87d3e55fd9147d51f3a522a558e96